plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MavenUtils.java

```diff
@@ -177,4 +180,18 @@ public static ArtifactRepository createRepository(String id, String url) throws
     return MavenPlugin.getMaven().createArtifactRepository(id, url);
   }
 
+  public static boolean isArtifactAvailableLocally(String groupId, String artifactId, String version) {
+    try {
+      Preconditions.checkArgument(!MAVEN_LATEST_VERSION.equals(version));
+      String artifactPath =
+          MavenPlugin.getMaven().getLocalRepository()
+          .pathOf(new DefaultArtifact(groupId, artifactId, version, null /* scope */, "jar",
+                                      null /*classifier */, new DefaultArtifactHandler("jar")));
+      return new File(artifactPath).exists();
+    } catch (CoreException ex) {
+      logger.log(Level.SEVERE, "Could not lookup local repository", ex);
```

Comment:
I mean, this is somewhat like `File.exists()`, in that it checks the availability of something that returns `boolean`. In the context of `File.exists()`, a non-existing file will be a serious error in a certain context like you said, but `File.exists()` doesn't log SEVERE whenever it returns false.

If `isArtifactAvailableLocally()` returning false is a serious error, perhaps names like `checkArtifactLocalAvailability()` is better suited, throwing an exception?